crates/fluvio/src/producer/partition_producer.rs

```diff
@@ -212,27 +239,42 @@ impl PartitionProducer {
         request.timeout = self.config.timeout;
         request.topics.push(topic_request);
 
-        let response = spu_socket.send_receive(request).await?;
+        let response = if self.client_stats.is_collect(ClientStatsDataCollect::Data) {
+            let (response, send_latency) = self
+                .client_stats
+                .send_and_measure_latency(&spu_socket, request)
+                .await?;
 
-        for (batch_notifier, partition_response) in batch_notifiers.into_iter().zip(
+            client_stats_update += send_latency;
             response
-                .responses
-                .into_iter()
-                .flat_map(|response| response.partitions),
-        ) {
-            let base_offset = partition_response.base_offset;
-            let fluvio_error = partition_response.error_code.clone();
+        } else {
+            spu_socket.send_receive(request).await?
+        };
+
+        let mut base_offset = 0;
+        for (batch_notifier, response) in batch_notifiers.into_iter().zip(response.responses.iter())
```

Comment:
could you change this to
```suggestion
         for (batch_notifier, partition_response) in batch_notifiers.into_iter().zip(responses.into_iter().flat_map(|response| response.partitions))
```